uikit/views/src/main/java/com/sygic/maps/uikit/views/searchresultlist/adapter/ResultListAdapter.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019 Sygic a.s. All rights reserved.
+ *
+ * This project is licensed under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.views.searchresultlist.adapter
+
+import android.os.Parcelable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.recyclerview.widget.RecyclerView
+import com.sygic.maps.uikit.views.R
+import com.sygic.maps.uikit.views.searchresultlist.data.SearchResultItem
+
+abstract class ResultListAdapter<P : Parcelable, T : ResultListAdapter.ItemViewHolder<P>> : RecyclerView.Adapter<T>() {
+
+    var clickListener: ClickListener<P>? = null
```

Comment:
ten listener je optional? ked nechces nenastavis a sme s tym ok hej?

Reply:
tak nikdo tě přeci nenutí si ho nastavit, nevidím důvod proč ho dávat required, dokáži si přestavit i neklikací použití. a my si checkujeme či je nastavený nebo není.

Reply:
to ano, ale vacsinou mam skor pocit, ze sa to dava do konstruktoru, kedze vacsina pouziti je s klikanim... tam by mohol byt samozrejme default `= null` ale mat to ako seter je trosku ine, lebo to mozes kedykolvek menit nastavovat a hlavne musis nezabudnut pri vytvarani, ze to chces nastavit. Z konstruktoru by to mohlo byt `val` `ClickListener<P>?`